src/libs/antares/study/scenario-builder/rules.cpp

```diff
@@ -39,62 +39,63 @@ namespace Data
 {
 namespace ScenarioBuilder
 {
-Rules::Rules() :
- load(), solar(), hydro(), wind(), thermal(), renewable(), hydroLevels(), pAreaCount(0)
+Rules::Rules(Study& study) : 
+    study_(study), load(), solar(), hydro(), wind(), thermal(), renewable(), hydroLevels(), pAreaCount(0)
 {
 }
 
 Rules::~Rules()
 {
     delete[] thermal;
+    delete[] renewable;
 }
 
-void Rules::saveToINIFile(const Study& study, Yuni::IO::File::Stream& file) const
+void Rules::saveToINIFile(Yuni::IO::File::Stream& file) const
 {
     file << "[" << pName << "]\n";
     if (pAreaCount)
     {
         // load
-        load.saveToINIFile(study, file);
+        load.saveToINIFile(study_, file);
         // solar
-        solar.saveToINIFile(study, file);
+        solar.saveToINIFile(study_, file);
         // hydro
-        hydro.saveToINIFile(study, file);
+        hydro.saveToINIFile(study_, file);
         // wind
-        wind.saveToINIFile(study, file);
+        wind.saveToINIFile(study_, file);
         // Thermal, each area
         for (uint i = 0; i != pAreaCount; ++i)
         {
-            thermal[i].saveToINIFile(study, file);
-            renewable[i].saveToINIFile(study, file);
+            thermal[i].saveToINIFile(study_, file);
+            renewable[i].saveToINIFile(study_, file);
         }
         // hydro levels
-        hydroLevels.saveToINIFile(study, file);
+        hydroLevels.saveToINIFile(study_, file);
```

Comment:
If `study` was set as a protected member of `TSNumberData`, you would not need to pass it here to methods of derived classes. See remark in **rules.h**.